### ISSUE TITLE
test student

### ISSUE THREAD

**Issue by loighic** [Carnap/Carnap#303]
Hi. I just noticed that the test student that I can enroll in my Carnap course isn't a "fresh" student for this semester's course. In particular, it seems to get the due dates from an earlier semester---but not the assignments from an earlier semester, I don't think. But I'm not certain about that. (I went back and removed the test student from the earlier semester, and so I can't test it further.) The "Book" link also didn't work properly until I deleted all previous test students and created a new one.

Anyway, removing the test student from the earlier course, seems to fix the problem, and so that's one solution. I don't know if there's any other way to just automatically get a new test student for each course?

**Comment by gleachkr** [comment]
Huh, puzzling. I'll see if I can reproduce, but I might need some more information.

1. Is the test student here a separate account, or is it the one that you access on the main account by looking at your user page?
2. How are you switching the test student to a new course?

The issue with the book link might be explained if the test student were the one associated with your main account, and if it wasn't fully switched over to the new course, for whatever reason - where the book link points can depend on what course this account is enrolled in.

About automatically getting a new student: it *should* be enough to migrate the student side of your main account to the new course, by editing "course enrollment" under "personal information" at the bottom of your user page - once you've switched to a new class, there shouldn't be any artifacts left from the old one, except for the record of any exercises in the default textbook that you've completed (this is a side effect of the quirky way that the textbook exercises are handled, one reason why we want to phase it out as default).

**Comment by loighic** [comment]
Yeah, I should have investigated more carefully before removing the test students from the earlier course. I don't think that I can reproduce it on my end without waiting another semester. 

(1) All that I have is the attached screenshot (which I got when I was trying to figure out what was going on with the book link). That's a screenshot of my current course, but with the Spring 2021 due dates. Right now, I just have one course in Carnap, and, except for briefly opening my Spring 2021 course to remove the test students from it, I haven't had any old courses open.

(2) Using the LTI tool in Canvas, I actually get two test students. One that is generated by accessing Carnap while I am in the "Student View" in Canvas, and one that is generated by accessing Carnap when I am in the regular, instructor view in Canvas. (I'm not sure if this is relevant to the current issue, but I thought that I'd mention it.) 

(3) I've never touched this: editing "course enrollment" under "personal information" at the bottom of my user page.

(4) To fix everything yesterday, I removed the test students from my Spring 2021 course, which was the first semester that I used the LTI tool. So, I presume that's part of this (?). 

In any case, this isn't really a problem for me now, especially for this semester. But I thought that I should put it in as an issue. Thanks.

![old_test_student](https://user-images.githubusercontent.com/54258105/159542660-46a3801b-9e6b-468a-9c4f-f61ce4276e58.png)
